Lib/profiling/sampling/gecko_collector.py

```diff
@@ -0,0 +1,451 @@
+import json
+import os
+import platform
+import sys
+import time
+from typing import Dict, List, Optional, Tuple, Any
+
+from .collector import Collector, THREAD_STATE_RUNNING
+
+
+class GeckoCollector(Collector):
+    def __init__(self, *, skip_idle=False):
+        self.skip_idle = skip_idle
+        self.start_time = time.time() * 1000  # milliseconds since epoch
+
+        # Global string table (shared across all threads)
+        self.global_strings = ["(root)"]  # Start with root
+        self.global_string_map = {"(root)": 0}
+
+        # Per-thread data structures
+        self.threads = {}  # tid -> thread data
+
+        # Global tables
+        self.libs = []
+
+        # Sampling interval tracking
+        self.sample_count = 0
+        self.last_sample_time = 0
+        self.interval = 1.0  # Will be calculated from actual sampling
+
+    def collect(self, stack_frames):
+        """Collect a sample from stack frames."""
+        current_time = (time.time() * 1000) - self.start_time
+
+        # Update interval calculation
+        if self.sample_count > 0 and self.last_sample_time > 0:
+            self.interval = (
+                current_time - self.last_sample_time
+            ) / self.sample_count
+        self.last_sample_time = current_time
+
+        for interpreter_info in stack_frames:
+            for thread_info in interpreter_info.threads:
+                if (
+                    self.skip_idle
+                    and thread_info.status != THREAD_STATE_RUNNING
+                ):
+                    continue
+
+                frames = thread_info.frame_info
+                if not frames:
+                    continue
+
+                tid = thread_info.thread_id
+
+                # Initialize thread if needed
+                if tid not in self.threads:
+                    self.threads[tid] = self._create_thread(tid)
+
+                thread_data = self.threads[tid]
+
+                # Process the stack
+                stack_index = self._process_stack(thread_data, frames)
+
+                # Add sample - cache references to avoid dictionary lookups
+                samples = thread_data["samples"]
+                samples["stack"].append(stack_index)
+                samples["time"].append(current_time)
+                samples["eventDelay"].append(None)
+
+        self.sample_count += 1
+
+    def _create_thread(self, tid):
```

Comment:
ℹ️ **Style**

Consider adding type annotations to the `_create_thread` method for better code readability and maintainability. This helps other developers understand the expected types of the parameters and return value.

**Suggested fix:**
```suggestion
def _create_thread(self, tid: int) -> Dict[str, Any]:
```